cmake/Modules/CodeCoverage.cmake

```diff
@@ -192,23 +192,26 @@ function(ADD_CODE_COVERAGE)
       set(OMIT_FLAGS --omit ${Coverage_EXCLUDES})
     endif()
 
+    add_custom_target(${Coverage_NAME}_nosetests_py
+        COMMAND ${PYTHON_COVERAGE_PATH} report --include "*${REAL_SOURCE_DIR}*" ${OMIT_FLAGS} || echo "WARNING: No python nosetests report to output"
+        COMMAND ${PYTHON_COVERAGE_PATH} xml  -o ${Coverage_NAME}_nosetests_python.xml --include "*${REAL_SOURCE_DIR}*" ${OMIT_FLAGS} || echo "WARNING: No python nosetests xml to output"
+        WORKING_DIRECTORY ${PROJECT_BINARY_DIR}
+        DEPENDS ${Coverage_NAME}_cpp
+    )
     # Create Python coverage report
     add_custom_target(${Coverage_NAME}_py
         # Rename .coverage file generated by nosetests to avoid overwriting during combine step
         COMMAND if [ -f ${PROJECT_BINARY_DIR}/.coverage ]\; then mv ${PROJECT_BINARY_DIR}/.coverage ${PROJECT_BINARY_DIR}/.coverage.nosetests\; fi
-        COMMAND cp ${PROJECT_BINARY_DIR}/.coverage* ${COVERAGE_DIR}/ || echo "WARNING: No python coverage!"
-        COMMAND ${PYTHON_COVERAGE_PATH} combine || echo "WARNING: No python coverage to combine!"
-        COMMAND ${PYTHON_COVERAGE_PATH} report --include "*${REAL_SOURCE_DIR}*" ${OMIT_FLAGS} || echo "WARNING: no python report to output"
+        COMMAND ${PYTHON_COVERAGE_PATH} combine || echo "WARNING: No python coverage to combine"
+        COMMAND ${PYTHON_COVERAGE_PATH} report --include "*${REAL_SOURCE_DIR}*" ${OMIT_FLAGS} || echo "WARNING: No python report to output"
         COMMAND ${PYTHON_COVERAGE_PATH} xml  -o ${Coverage_NAME}_python.xml --include "*${REAL_SOURCE_DIR}*" ${OMIT_FLAGS} || echo "WARNING: No python xml to output"
-        COMMAND ${PYTHON_COVERAGE_PATH} html -d ${Coverage_NAME}_python_html --include "*${REAL_SOURCE_DIR}*" ${OMIT_FLAGS} || echo "WARNING: No python html to output"
-        COMMAND cp ${COVERAGE_DIR}/${Coverage_NAME}_python.xml ${PROJECT_BINARY_DIR} || echo "WARNING: No python xml to copy"
-        COMMAND cp -r ${COVERAGE_DIR}/${Coverage_NAME}_python_html ${PROJECT_BINARY_DIR} || echo "WARNING: No python html to copy"
+        COMMAND cp ${COVERAGE_DIR}/${Coverage_NAME}_python.xml ${PROJECT_BINARY_DIR}/ || echo "WARNING: No python xml"
```

Comment:
Don't we need to list this xml in OUTPUT?

Reply:
i change to use `add_custom_command`